consensus/ballot_state.go

```diff
@@ -268,7 +269,7 @@ func (s *BallotState) MaybeConfirmAsPrepared(n int, x SlotValue) bool {
 
 	s.Logf("confirms as prepared: %s", &Ballot{n: n, x: x})
 	
-	if s.hn == n && *s.z != x {
```

Comment:
why did you remove this check, is it just redundant?

Reply:
Yeah